django/contrib/contenttypes/checks.py

```diff
@@ -18,3 +19,21 @@ def check_generic_foreign_keys(app_configs=None, **kwargs):
     for field in fields:
         errors.extend(field.check())
     return errors
+
+
+def check_model_name_lengths(app_configs=None, **kwargs):
+    if app_configs is None:
+        models = apps.get_models()
+    else:
+        models = chain.from_iterable(app_config.get_models() for app_config in app_configs)
+
+    errors = []
+    for model in models:
+        if len(model._meta.model_name) > 100:
+            errors.append(Error(
+                'Model name is too long',
```

Comment:
I would say, 'Model names must be at most 100 characters (got %d).' I don't think a hint is needed.